core/primitives/src/transaction.rs

```diff
@@ -214,6 +215,12 @@ impl BorshDeserialize for Transaction {
     }
 }
 
+pub enum BatchValidationError {
+    InvalidTx(InvalidTxError), // Individual transaction failed a non-signature check
```

Comment:
nit: consider using `/// comment` to have doc comments on variants, for nicer integration with IDEs and generated docs